packages/core/src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -505,6 +509,8 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
                 throw new Error(result.error.message);
             }
 
+            const { id: token } = catchedConfirmError || result == null ? { id: clientSecret } : result?.paymentIntent || { id: '' };
```

Comment:
```suggestion
            const { id: token } = catchedConfirmError || result === null ? { id: clientSecret } : result?.paymentIntent || { id: '' };
```

Reply:
Also as a result of this statement does it mean we will have a value of token based on conditionals? 

Is it better to move this logic to another function, feels like a bit too much happening in ternary conditions.